pkg/cmd/provisioning/ibmcloud/refresh-keys.go

```diff
@@ -0,0 +1,159 @@
+	"fmt"
+	"log"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/openshift/cloud-credential-operator/pkg/ibmcloud"
+)
+
+// NewRefreshKeysCmd provides the "refresh-keys" subcommand
+func NewRefreshKeysCmd() *cobra.Command {
+	refreshKeysCmd := &cobra.Command{
+		Use:   "refresh-keys",
+		Short: "Refresh API Keys for the Service ID",
+		RunE:  refreshKeysCmd,
+	}
+
+	refreshKeysCmd.PersistentFlags().StringVar(&Options.Name, "name", "", "User-defined name for all created IBM Cloud resources (can be separate from the cluster's infra-id)")
+	refreshKeysCmd.MarkPersistentFlagRequired("name")
+	refreshKeysCmd.PersistentFlags().StringVar(&Options.CredRequestDir, "credentials-requests-dir", "", "Directory containing files of CredentialsRequests to delete IAM Roles for (can be created by running 'oc adm release extract --credentials-requests --cloud=ibmcloud' against an OpenShift release image)")
+	refreshKeysCmd.MarkPersistentFlagRequired("credentials-requests-dir")
+	refreshKeysCmd.PersistentFlags().BoolVar(&Options.Force, "force", false, "delete all the service account forcefully(will delete all the entries with the name)")
+	refreshKeysCmd.PersistentFlags().StringVar(&Options.KubeConfigFile, "kubeconfig", "", "absolute path to the kubeconfig file")
+	refreshKeysCmd.MarkPersistentFlagRequired("kubeconfig")
+	refreshKeysCmd.PersistentFlags().StringVar(&Options.ResourceGroupName, "resource-group-name", "", "Name of the resource group used for scoping the access policies")
+	refreshKeysCmd.PersistentFlags().BoolVar(&Options.Create, "create", false, "Create the ServiceID if does not exists")
+
+	return refreshKeysCmd
+}
+
+func refreshKeysCmd(cmd *cobra.Command, args []string) error {
+	apiKey := getEnv(APIKeyEnvVars)
+	if apiKey == "" {
+		return fmt.Errorf("%s environment variable not set", APIKeyEnvVars)
+	}
+
+	params := &ibmcloud.ClientParams{
+		InfraName: Options.Name,
+	}
+
+	ibmclient, err := ibmcloud.NewClient(apiKey, params)
+	if err != nil {
+		return err
+	}
+
+	apiKeyDetailsOptions := ibmclient.NewGetAPIKeysDetailsOptions()
+	apiKeyDetailsOptions.SetIamAPIKey(apiKey)
+	apiKeyDetails, _, err := ibmclient.GetAPIKeysDetails(apiKeyDetailsOptions)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get Details for the given APIKey")
+	}
+
+	cs, err := newClientset(Options.KubeConfigFile)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create the kubernetes clientset")
+	}
+	err = refreshKeys(ibmclient, cs, apiKeyDetails.AccountID, Options.Name, Options.ResourceGroupName, Options.CredRequestDir, Options.Create)
+	if err != nil {
+		return errors.Wrap(err, "Failed to refresh keys")
+	}
+	return nil
+}
+
+func refreshKeys(client ibmcloud.Client, cs *kubernetes.Clientset, accountID *string, name, resourceGroupName, credReqDir string, create bool) error {
```

Comment:
can we give the client parameters more distinct names? something like this (or really just anything more descriptive than client and cs)
```suggestion
func refreshKeys(ibmcloudClient ibmcloud.Client, kubeClient *kubernetes.Clientset, accountID *string, name, resourceGroupName, credReqDir string, create bool) error {
```